build-tools/automation/azure-pipelines.yaml

```diff
@@ -0,0 +1,213 @@
+        msbuildArguments: /t:ZipBuildStatus /p:ResultPackageOutputPath=$(Build.ArtifactStagingDirectory)
+      condition: always()
+
+    - task: PublishPipelineArtifact@0
+      displayName: upload artifacts
+      inputs:
+        artifactName: mac-build-results
+        targetPath: $(Build.ArtifactStagingDirectory)
+      condition: always()
+
+- stage: win_build_test
+  displayName: Win
+  dependsOn: prepare
+  jobs:
+  - job: build_test
+    displayName: Build and Test
+    pool: VSEng-MicroBuildVS2017
+    timeoutInMinutes: 360
+    cancelTimeoutInMinutes: 5
+    steps:
+    - task: DownloadPipelineArtifact@1
+      inputs:
+        artifactName: $(BundleArtifactName)
+        downloadPath: $(System.DefaultWorkingDirectory)
+
+    - task: MSBuild@1
+      displayName: msbuild Xamarin.Android /t:Prepare
+      inputs:
+        solution: Xamarin.Android.sln
+        configuration: $(XA.Build.Configuration)
+        msbuildArguments: /t:Prepare /bl:bin\Build$(Configuration)\msbuild-prepare.binlog /p:BundleRootPath=$(System.DefaultWorkingDirectory)
+
+    - task: MSBuild@1
+      displayName: msbuild Xamarin.Android /t:Build
+      inputs:
+        solution: Xamarin.Android.sln
+        configuration: $(XA.Build.Configuration)
+        msbuildArguments: /t:Build /bl:bin\Build$(Configuration)\msbuild-build.binlog /p:BundleRootPath=$(System.DefaultWorkingDirectory)
+
+    - task: MSBuild@1
+      displayName: msbuild create-vsix
+      inputs:
+        solution: build-tools\create-vsix\create-vsix.csproj
+        configuration: $(XA.Build.Configuration)
+        msbuildArguments: /p:CreateVsixContainer=True /p:ZipPackageCompressionLevel=Normal /bl:bin\Build$(Configuration)\msbuild-create-vsix.binlog
+
+    - task: CmdLine@1
+      displayName: xabuild Xamarin.Android-Tests
+      inputs:
+        filename: bin\$(XA.Build.Configuration)\bin\xabuild.exe
+        arguments: Xamarin.Android-Tests.sln /p:Configuration=$(XA.Build.Configuration) /p:XAIntegratedTests=False /bl:bin\Build$(Configuration)\msbuild-build-tests.binlog
+
+    - task: MSBuild@1
+      displayName: nunit Xamarin.Android.Build.Tests
+      inputs:
+        solution: Xamarin.Android.sln
+        configuration: $(XA.Build.Configuration)
+        msbuildArguments: /t:RunNUnitTests /bl:bin\Build$(Configuration)\msbuild-run-nunit-tests.binlog
+      timeoutInMinutes: 240
+
+    - task: MSBuild@1
+      displayName: nunit Java.Interop Tests
+      inputs:
+        solution: Xamarin.Android.sln
+        configuration: $(XA.Build.Configuration)
+        msbuildArguments: >
+          /t:RunJavaInteropTests
+          /p:TestAssembly="bin\Test$(XA.Build.Configuration)\generator-Tests.dll;bin\Test$(XA.Build.Configuration)\Java.Interop.Tools.JavaCallableWrappers-Tests.dll;bin\Test$(XA.Build.Configuration)\LogcatParse-Tests.dll;bin\Test$(XA.Build.Configuration)\Xamarin.Android.Tools.ApiXmlAdjuster-Tests.dll;bin\Test$(XA.Build.Configuration)\Xamarin.Android.Tools.Bytecode-Tests.dll"
```

Comment:
Why override `TestAssembly`?

I also don't think this works like this implies?  `@(_TestAssembly)` is an item group, not a property; afaik, `/p:` only impacts properties, not item groups.  I thus believe that this line will have *no impact at all*.

I don't see *any* meaningful instances of `(TestAssembly`:

```shell
$ git grep '(TestAssembly'
tests/TestRunner.Core/TestInstrumentation.cs:                   if (TestAssemblyDirectories == null || TestAssemblyDirectories.Count == 0)
tests/TestRunner.Core/TestInstrumentation.cs:                   if (TestAssemblyDirectories != null && TestAssemblyDirectories.Count > 0) {
tests/TestRunner.NUnit/NUnitTestRunner.cs:                      foreach (TestAssemblyInfo assemblyInfo in testAssemblies) {
tests/TestRunner.xUnit/XUnitTestRunner.cs:                      foreach (TestAssemblyInfo assemblyInfo in testAssemblies) {
tests/TestRunner.xUnit/XUnitTestRunner.cs:              protected virtual ITestFrameworkDiscoveryOptions GetFrameworkOptionsForDiscovery (TestAssemblyConfiguration configuration)
tests/TestRunner.xUnit/XUnitTestRunner.cs:              protected virtual ITestFrameworkExecutionOptions GetFrameworkOptionsForExecution (TestAssemblyConfiguration configuration)
```

Reply:
This is something that was copied directly out of the existing Windows build, perhaps @jonathanpeppers can also help determine whether or not this property override is needed?

Reply:
@jonpryor I think we can fix this one up in a different PR.

1. We need to change Java.Interop to list which tests assemblies work on Windows: https://github.com/xamarin/java.interop/blob/7ef0be5ebd7e9e27b679aadfab8d488029ea64f2/build-tools/scripts/RunNUnitTests.targets#L17-L18 (at some point we could actually make them all work on Windows if we could build java-interop.dll for Windows)
2. We should port [this Java.Interop job](http://build.devdiv.io/7515) to Azure Pipelines YAML while we're at it. It currently lists all the test assemblies, too.
3. We could then drop this line from Xamarin.Android's YAML completely

Reply:
As @jonathanpeppers pointed out it looks like this is [used here](https://github.com/xamarin/java.interop/blob/3754536891a5b6d8b033a34bcfa23e39984ae925/build-tools/scripts/RunNUnitTests.targets#L18) to filter out some suites which won't work against this Windows build.